SocketChannelTest.testConnect flaky

```
Test Case 'SocketChannelTest.testConnect' started at 2019-04-12 10:06:47.202
/Users/johannes/.Trash/.nio-release-tools_5zt3xq/swift-nio/Tests/NIOTests/SocketChannelTest.swift:246: error: SocketChannelTest.testConnect : XCTAssertNoThrow failed: threw error "alreadyClosed" - 
Test Case 'SocketChannelTest.testConnect' failed (0.007 seconds)
```

on commit ba7970fe396e8198b84c6c1b44b38a1d4e2eb6bd (`nio-1.14` branch but will very likely affect 2.0 too)

oddly failed in 2.0 too at the same time

```
Test Case 'SocketChannelTest.testConnect' started at 2019-04-12 10:06:47.202
/Users/johannes/.Trash/.nio-release-tools_5zt3xq/swift-nio/Tests/NIOTests/SocketChannelTest.swift:246: error: SocketChannelTest.testConnect : XCTAssertNoThrow failed: threw error "alreadyClosed" - 
Test Case 'SocketChannelTest.testConnect' failed (0.007 seconds)
```

both in Docker, so possibly it was a docker VM networking issue

still an issue

```
19:25:12 Test Case 'SocketChannelTest.testConnect' started at 2019-11-27 19:25:12.792
19:25:12 /code/Tests/NIOTests/SocketChannelTest.swift:246: error: SocketChannelTest.testConnect : XCTAssertNoThrow failed: threw error "alreadyClosed" - 
19:25:12 Test Case 'SocketChannelTest.testConnect' failed (0.005 seconds)
```

another one:

```
14:22:00 Test Case 'SocketChannelTest.testConnect' started at 2019-12-06 14:22:00.658
14:22:00 /code/Tests/NIOTests/SocketChannelTest.swift:246: error: SocketChannelTest.testConnect : XCTAssertNoThrow failed: threw error "alreadyClosed" - 
14:22:00 Test Case 'SocketChannelTest.testConnect' failed (0.004 seconds)
```

and again :(

```
15:12:56 Test Case 'SocketChannelTest.testConnect' started at 2019-12-17 15:12:56.002
15:12:56 /code/Tests/NIOTests/SocketChannelTest.swift:246: error: SocketChannelTest.testConnect : XCTAssertNoThrow failed: threw error "alreadyClosed" - 
15:12:56 Test Case 'SocketChannelTest.testConnect' failed (0.005 seconds)
```

--- a/Tests/NIOTests/SocketChannelTest.swift
+++ b/Tests/NIOTests/SocketChannelTest.swift
@@ -237,12 +237,16 @@ public final class SocketChannelTest : XCTestCase {
                                                                eventLoop: eventLoop as! SelectableEventLoop))
         let promise = channel.eventLoop.makePromise(of: Void.self)
 
-        XCTAssertNoThrow(try channel.pipeline.addHandler(ActiveVerificationHandler(promise)).flatMap {
-            channel.register()
-        }.flatMap {
-            channel.connect(to: try! SocketAddress(ipAddress: "127.0.0.1", port: 9999))
-        }.flatMap {
-            channel.close()
+        XCTAssertNoThrow(try channel.eventLoop.flatSubmit {
+            // We need to hop to the EventLoop here to make sure that we don't get an ECONNRESET before we manage
+            // to close.
+            channel.pipeline.addHandler(ActiveVerificationHandler(promise)).flatMap {
+                channel.register()
+            }.flatMap {
+                channel.connect(to: try! SocketAddress(ipAddress: "127.0.0.1", port: 9999))
+            }.flatMap {
+                channel.close()
+            }
         }.wait())
 
         XCTAssertNoThrow(try channel.closeFuture.wait())

